src/imitation/scripts/config/train_adversarial.py

```diff
@@ -102,6 +102,35 @@ def seals_ant():
 
 @train_adversarial_ex.named_config
 def half_cheetah():
+    locals().update(**MUJOCO_SHARED_LOCALS)
```

Comment:
A shame there's so much duplication between this and the seals HalfCheetah config below. Factor out common config parameters (I think everything except `common=dict(env_name=...)`) into a dictionary e.g. `HALFCHEETAH_SHARED_LOCALS` and use the `locals().update` trick?

Reply:
I implemented the variant with the `locals().update()` for now for the sake of not expanding the scope of this PR any further.

However I don't like it since it is depends on knowing the smart hack that sacred uses to implement configs as functions. I feel it would be more hones to just use

```python
train_adversarial_ex.add_named_config("seals_half_cheetah", **CHEETAH_CONFIG, common = dict(env_name="HalfCheetah-v2"))
```

instead.

Reply:
Using `locals().update()` just sets the local variables the same as a direct variable definition. So in that sense I don't think it requires any knowledge of Sacred internals beyond "Sacred somehow extracts the local variables" -- which is already needed to understand the function definition style in the first place.

I do think there's a case to be made for using `add_named_config` over function decorator as the default if we're not doing any actual computation in the function, as I agree the function decorator approach is unnecessarily magic. But IMO  if we do this we should make the change everywhere we can, not just for `locals().update`. Probably best addressed in a separate.